src/java.management/share/classes/sun/management/spi/PlatformMBeanProvider.java

```diff
@@ -222,13 +218,4 @@ private PlatformMBeanProvider(Void unused) {
      * MBeans provided by this provider.
      */
     public abstract List<PlatformComponent<?>> getPlatformComponentList();
-
-    private static Void checkSubclassPermission() {
-        @SuppressWarnings("removal")
-        SecurityManager sm = System.getSecurityManager();
-        if (sm != null) {
-            sm.checkPermission(new RuntimePermission(PlatformMBeanProvider.class.getName()+".subclass"));
-        }
-        return null;
-    }
```

Comment:
I have verified in module-info.java that sun.management.spi is only conditionally exported so I agree that the explicit permission check can now be removed.

Reply:
thanks